code/modules/mob/living/carbon/human/species.dm

```diff
@@ -542,27 +542,26 @@ GLOBAL_LIST_EMPTY(features_by_species)
  * Proc called when mail goodies need to be updated for this species.
  *
  * Updates the mail goodies if that is required. e.g. for the blood deficiency quirk, which sends bloodbags to quirk holders, update the sent bloodpack to match the species' exotic blood.
- * Add implementation as needed for each individual species
+ * Add implementation as needed for each individual species. The base species proc should give the species the 'default' version of whatever mail goodies are required.
  * Arguments:
  * * mob/living/carbon/human/recipient - the mob receiving the mail goodies
- * * list/mail_goodies - a list of mail goodies
+ * * list/mail_goodies - a list of mail goodies. You should not be using this argument on the initial function call unless there is no other choice. You should instead add to the species' implementation of this proc.
  */
 /datum/species/proc/update_mail_goodies(mob/living/carbon/human/recipient, list/mail_goodies)
 	var/datum/quirk/blooddeficiency/blooddeficiency = recipient.get_quirk(/datum/quirk/blooddeficiency)
-	if(isnull(blooddeficiency))
-		return
-	if(!isnull(mail_goodies))
-		blooddeficiency.mail_goodies = mail_goodies
-	else
-		// no blood packs should be sent if mob has TRAIT_NOBLOOD and no exotic blood (like if a mob transforms into a plasmaman)
-		if(HAS_TRAIT(recipient, TRAIT_NOBLOOD) && isnull(recipient.dna.species.exotic_blood))
-			blooddeficiency.mail_goodies = null
+	
+	if(!isnull(blooddeficiency))
+		if(isnull(mail_goodies)) // The default case, e.g. species has no specific implementation
```

Comment:
`mail_goodies` is an empty list by default, no? why should we mutate it into a null, let's just keep an empty list an empty list

```suggestion
		if(!length(mail_goodies)) // The default case, e.g. species has no specific implementation
```